orttraining/orttraining/core/optimizer/memory_optimizer/recompute_analysis.cc

```diff
@@ -19,7 +19,7 @@ namespace onnxruntime::optimizer::memory_optimizer {
 
 namespace {
 
-constexpr int32_t MAXIMUM_RECOMPUTE_NODE_COUNT = 15;
+constexpr int32_t MAXIMUM_RECOMPUTE_NODE_COUNT = 50;
```

Comment:
how do we decide 50?

Reply:
Good question! honestly speaking, it is a number for more than a couple of nodes but not too many, so choose this. For one transformer layers, we can estimated there are a couple of major compute nodes, plus some cheap to compute nodes for example Shape, Slice, in the rotary embedding usage. So 15 is not enough for existing one layer, so increase to a number I feel bigger than 15, but not that many (while if there are too many, it means we need more fusion work).